inc/sysrepo.h

```diff
@@ -172,15 +204,24 @@ typedef struct sr_node_s {
 
     /**
      * Name of the module that defines scheme of this node.
-     * If it is NULL, the module name is inherited from the parent node.
+     * NULL if it is the same as that of the predecessor.
      */
     char *module_name;
 
-    /** Array of node's direct descendants. */
-    struct sr_node_s *children;
+    /** Pointer to the parent node (NULL in case of root node). */
+    struct sr_node_s *parent;
+
+    /** Pointer to the next sibling node (NULL if there is no one). */
+    struct sr_node_s *next;
+
+    /** Pointer to the previous sibling node (NULL if there is no one). */
+    struct sr_node_s *prev;
+
+    /** Pointer to the first child node (NULL if this is a leaf). */
```

Comment:
maybe we could also provide some public iterator macro for convenient iterating over the children?